lib/src/breakpoints_utils.dart

```diff
@@ -1,40 +1,65 @@
-import 'package:adaptive_breakpoints/adaptive_breakpoints.dart';
-
 import 'package:flutter/material.dart';
+import 'package:flutter_adaptive_scaffold/flutter_adaptive_scaffold.dart';
 
-extension Breakpoints on BuildContext {
-  /// All the breakpoints according to Material Design's breakpoint system.
-  BreakpointSystemEntry get breakpoints => getBreakpointEntry(this);
-
-  /// Double the default gutter size.
-  double get gutterLarge => (_material3Value ?? breakpoints.gutter) * 2;
-
-  /// The gutter size according to Material Design's breakpoints system.
-  double get gutter => _material3Value ?? breakpoints.gutter;
-
-  /// Half the default gutter size.
-  double get gutterSmall => (_material3Value ?? breakpoints.gutter) / 2;
-
-  /// A quarter the default gutter size.
-  double get gutterTiny => (_material3Value ?? breakpoints.gutter) / 4;
+import '../flutter_gutter.dart';
 
+/// A gap of the standard gutter size according to Material Design's breakpoints
+extension BreakpointExtensions on BuildContext {
   /// The margin size according to Material Design's breakpoints system.
-  double get margin => _material3Value ?? breakpoints.margin;
-
-  int get columns => breakpoints.columns;
+  double get margin => materialSpacing;
 
-  double? get _material3Value {
-    if (!Theme.of(this).useMaterial3) {
-      return null;
+  /// The size according to Material Design's with a specified [GutterType].
+  double gutter({GutterType type = GutterType.medium, double? scaleFactor}) {
+    switch (type) {
+      case GutterType.tiny:
+        return materialSpacing / (scaleFactor ?? 4);
+      case GutterType.small:
+        return materialSpacing / (scaleFactor ?? 2);
+      case GutterType.medium:
+        return materialSpacing;
+      case GutterType.large:
+        return materialSpacing * (scaleFactor ?? 2);
+      case GutterType.extraLarge:
+        return materialSpacing * (scaleFactor ?? 4);
+      case GutterType.expand:
+        return double.infinity;
     }
-    final Size size = MediaQuery.of(this).size;
-    if (size.width < 600 || size.height < 600) {
+  }
+
+  /// The size according to Material Design's breakpoints system.
+  double get materialSpacing {
+    if (Breakpoints.small.isActive(this)) {
       // Use the compact screen values.
       // See: https://m3.material.io/foundations/layout/applying-layout/compact
       return 16;
+    } else if (Breakpoints.medium.isActive(this)) {
+      // Use the medium/expanded screen values.
+      // See: https://m3.material.io/foundations/layout/applying-layout/medium
+      return 24;
+    } else if (Breakpoints.large.isActive(this)) {
+      // Use the large screen values.
+      // See: https://m3.material.io/foundations/layout/applying-layout/large-extra-large
+      return 24;
+    } else {
+      return 16;
+    }
+  }
+}
+
+/// Extension to add a gutter between the items of an iterable.
+extension SeparatedIterable on Iterable<Widget> {
```

Comment:
I'd lean towards not including this, especially given that the following looks like it's going to land:
https://github.com/flutter/flutter/pull/152472

It'd be useful in the meantime, but when it becomes obsolete it'd be best to deprecate it which will be annoying for everyone who started using it. Better to just not expose it in the first place IMO, but if you have strong feelings feel free to leave it in,

Reply:
Yeah i am aware of that one. It can still take a long time and is missing some features that we have in here now. I would keep this in for now and in a feature release make a 3.0 version or something to remove it.